modules/dialog/dlg_db_handler.c

```diff
@@ -905,19 +1002,21 @@ static str* write_dialog_vars( struct dlg_val *vars)
 	static int o_l=0;
 	struct dlg_val *v;
 	unsigned int l,i;
+	static str dst = {NULL,0};
 	char *p;
-
 	/* compute the required len */
-	for ( v=vars,l=0 ; v ; v=v->next) {
+	for ( v=vars,l=0; v ; v=v->next) {
 		l += v->name.len + 1 + v->val.len + 1;
```

Comment:
The code below is no longer necessary, since the control chars will be encoded anyway.